launch_ros/launch_ros/actions/lifecycle_transition.py

```diff
@@ -0,0 +1,257 @@
+                except ValueError:
+                    pass
+                finally:
+                    pass
+
+    def execute(
+        self,
+        context: launch.LaunchContext
+    ) -> Optional[List[Action]]:
+        """
+        Execute the Lifecycle Transition action.
+
+        :return Returns a list of actions to be executed to achieve specified transitions.
+          These are EventHandlers and EventEmitters for ChangeState and
+          StateTransition events of the nodes indicated.
+        """
+        lifecycle_node_names = [
+            perform_substitutions(context, name)
+            for name in self.__lifecycle_node_names]
+        subs_transition_ids = [
+            perform_substitutions(context, id)
+            for id in self.__transition_ids]
+        transition_ids = []
+        for id in subs_transition_ids:
+            transition_ids.append(int(id))
+
+        emit_actions = OrderedDict()
+        actions: List[Action] = []
+
+        # Create EmitEvents for ChangeStates and store
+        for name in lifecycle_node_names:
+            own_emit_actions = []
+            for id in transition_ids:
+                change_event = ChangeState(
+                    lifecycle_node_matcher=matches_node_name(name),
+                    transition_id=id)
+                emit_action = EmitEvent(
+                    event=change_event
+                )
+                own_emit_actions.append(emit_action)
+            emit_actions[name] = own_emit_actions
+            self.__event_handlers[name] = []
+        # Create Transition EventHandlers and Registration actions
+        i = 1
+        for id in transition_ids:
+            # Create Transition handler for all indicated nodes
+            for node_name in lifecycle_node_names:
+
+                states = self.transition_targets[id]
+                event_handler = None
+                # For all transitions except the last, emit next ChangeState Event
+                if i < len(transition_ids):
+                    event_handler = OnStateTransition(
+                        matcher=match_node_name_start_goal(
+                            node_name,
+                            states["start_state"],
+                            states["goal_state"]),
+                        entities=[
+                            emit_actions[node_name][i]],
+                        handle_once=True
+                    )
+                # For last transition emit Log message
+                else:
+                    event_handler = OnStateTransition(
+                        matcher=match_node_name_start_goal(
+                            node_name,
+                            states["start_state"],
+                            states["goal_state"]),
+                        entities=[
+                            launch.actions.OpaqueFunction(
+                                function=functools.partial(
+                                    self._remove_event_handlers,
```

Comment:
I don't think we need to explicitly remove event handlers in this case since we're using `handle_once=True` (which should automatically unregistered the handler).

Reply:
This should unregister the error handlers that were not triggered in this case.

Reply:
Makes sense. Maybe it's worth adding a brief comment here mentioning that we want to unregister the error event handlers that are registered below.